src/domain/tradingview/useTVDatafeed.ts

```diff
@@ -0,0 +1,97 @@
+import { getTokens } from "config/tokens";
+import { SUPPORTED_RESOLUTIONS } from "config/tradingview";
+import { timezoneOffset } from "domain/prices";
+import { useChainId } from "lib/chains";
+import { useEffect, useMemo, useRef } from "react";
+import { TVRequests } from "./TVRequests";
+
+const configurationData = {
+  supported_resolutions: Object.keys(SUPPORTED_RESOLUTIONS),
+  supports_marks: false,
+  supports_timescale_marks: false,
+  supports_time: true,
+  reset_cache_timeout: 100,
+};
+
+export default function useTVDatafeed() {
+  const { chainId } = useChainId();
+  const intervalRef = useRef<ReturnType<typeof setInterval> | null>();
```

Comment:
i think it would return undefined by default, not null. so correct type would be `ReturnType<typeof setInterval> | undefined>`